### PR TITLE
docs: adding architecture and components reference

### PR DESCRIPTION
**Pull request by salasberryfin**
### Issue #17 

This is an initial proposal for architecture and components documentation.

A new section is created `Reference Guides` following the structure used in [Rancher Docs](https://ranchermanager.docs.rancher.com/pages-for-subheaders/rancher-manager-architecture) containing an `Architecture` sub-section divided in:
- Introduction.
- Components.
- Deployment Scenarios.

~~The current docs use the diagrams from the Rancher Turtles RFC, which may need to be updated to accurately show the status of the first release since some of the features are not yet implemented. **Should the docs be strictly aligned to what we currently have in the early access release or should we show the architecture as a whole with details of what's to be implemented in future versions?**~~

Diagrams are now updated to show the current status.

Any ideas, edits and proposals are welcome. This should serve only as a starting point.

### CODE REVIEW

File: docs/reference-guides/architecture/in_cluster_topology.png
Comment (Danil-Grigorev) [inline docs/reference-guides/architecture/in_cluster_topology.png:1]: I think this diagram should be simplified for the initial release and exclude parts we don’t currently support. Like sync creds or UI parts, we don’t have them now.

**Comment by salasberryfin** [comment]
This is now ready to be reviewed. As proposed by @Danil-Grigorev and with the help of @richardcase, the diagrams are now updated to the current status.

File: docs/intro.md [hunk 1]
```diff
@@ -8,4 +8,10 @@ Welcome to the Rancher Turtles documentation site.
...
 
-Please checkout out [getting started guide](./getting-started/intro).
\ No newline at end of file
+Please checkout out [getting started guide](./getting-started/intro).
+
+## Reference Guides
+
+This section focuses on implementation details, including 
+[architecture](./reference-guides/architecture/intro) and how Rancher Turtles 
+integrates with Rancher.
```
Comment (richardcase) [inline docs/intro.md:17]:
```suggestion
This section focuses on implementation details, including 
[architecture](./reference-guides/architecture/intro) and how Rancher Turtles integrates with Rancher.
```

Comment (salasberryfin) [reply docs/intro.md]: Updated on last push.

**Review by Danil-Grigorev — APPROVED** [review]

**Review by richardcase — APPROVED** [review]

**Pull request merged**